src/devices/cpu/z180/z180asci.h

```diff
@@ -0,0 +1,73 @@
+// license:BSD-3-Clause
+// copyright-holders:Miodrag Milanovic
+/*********************************************************************
+
+    z180asci.h
+
+
+*********************************************************************/
+
+#ifndef MAME_CPU_Z180_Z180ASCI_H
+#define MAME_CPU_Z180_Z180ASCI_H
+
+#pragma once
+
+#include "diserial.h"
+
+class z180asci_channel :  public device_t,
+	public device_buffered_serial_interface<4U>
```

Comment:
Is this really a good case for `device_buffered_serial_interface` or do we have sufficient information on how it works that we can simulate it better with a full implementation?  `device_serial_interface` is notoriously bad for reproducing real-world UART behaviour properly.

Reply:
There is good documentation,  but implementing it using device_buffered_serial_interface was fitting nice, also a reason why channel is split device and not done in cpu implementation itself